cmod/board.py

```diff
@@ -1,194 +1,297 @@
+        file and the required format and try again.""")
+       self.clear()
+       return False
+
+ def get_detector(self, detid):
+   # -1 as detid is the detector's index in the list + 1
+   return self.detectors[detid-1]
+
+
+ def get_all_detectors(self):
+   return self.detectors
+ 
+ def set_id(self, id):
+   self.id = id
+
+
+#   def calib_dets(self):
+#    return sorted([k for k in self.get_all_detectors() if int(k) < 0], reverse=True)
+
+
+#    def add_calib_det(self, detid, mode=-1, channel=-1):
+#     if detid not in self.get_all_detectors() and int(detid) < 0:
+#         self.detectors[detid] = Detector({
+#             "mode": mode,
+#             "channel": channel,
+#             "default coordinates": [-100, -100]
+#         }, self)
+
+ # Get/Set calibration measures with additional parsing
+#   TODO: revisit while implementing conditions
+ def add_vis_coord(self, detid, z, data, filename):
+   self.detectors[detid-1]['coordinates']['calibrated'].append({
+     'command': 'visualcenterdet',
+     z: self.roundz(z),
+     'data': {
+       'coordinates': data,
+       'file': filename
+     }
+    })
+   
+   self.save_board()
+
+
+ def add_visM(self, detid, z, data, filename):
+   self.detectors[detid-1]['coordinates']['calibrated'].append({
+     'command': 'visualhscan',
+     'z': self.roundz(z),
+     'data': {
+        'transform': data,
+        'file': filename
+     }
+    })
+
+   self.save_board()
+
+
+
+ def add_lumi_coord(self, detid, z, data):
+   self.detectors[detid-1]['coordinates']['calibrated'].append({
+     'command': 'halign',
+     z: self.roundz(z),
+     'data': {
+       'coordinates': data
+     }
+    })
+
+   self.save_board()
+
+
+ def get_latest_entry(self, detid, commandname, z=None):
+   for i in range(len(self.detectors[detid-1]['coordinates']['calibrated'])-1, -1, -1):
+     entry = self.detectors[detid-1]['coordinates']['calibrated'][i]
+     if entry['command'] == commandname and (z is None or entry['z'] == self.roundz(z)):
+       return entry
+   return None
+
+ def get_vis_coord(self, detid, z):
+   return self.get_latest_entry(detid, 'visualcenterdet', z)
+
+
+ def get_visM(self, detid, z):
+   return self.get_latest_entry(detid, 'visualhscan', z)
+
+ def get_lumi_coord(self, detid, z):
+   return self.get_latest_entry(detid, 'halign', z)
+
+ def add_lumi_vis_separation(self, detid, z, h):
+   self.detectors[detid-1]['coordinates']['calibrated'].append({
+     'command': 'lumi_vis_separation', 
+     'z': self.roundz(z), 
+     'data': {
+       'separation': h
+     }
+    })
+   
+   self.save_board()
+
+
+ def vis_coord_hasz(self, detid, z):
+   return self.get_latest_entry(detid, 'visualcenterdet', z) is not None
+
+
+ def visM_hasz(self, detid, z):
+   return self.get_latest_entry(detid, 'visualhscan', z) is not None
+
+
+ def lumi_coord_hasz(self, detid, z):
+   return self.get_latest_entry(detid, 'halign', z) is not None
+
+
+# TODO: why is this needed??
```

Comment:
This is probably a bad name for the function. It should check if there are any calibration results for this tileboard. 

This method is required to so that the CLI and detect that when attempting to load a new tileboard into the session, and remind the user whether this tileboard contains calibration information and to make sure to save the results before continuing.